docs/security/access-control.md

```diff
@@ -507,6 +620,89 @@ boolean deleted =
 
 ## Permission Operation
 
+### Grant privileges to a role
+
+You can revoke specific privileges from a role.
+The request path for REST API is `/api/metalakes/{metalake}/permissions/roles/{role}/{metadataObjectType}/{metadataObjectName}/revoke`.
+
+<Tabs groupId='language' queryString>
+<TabItem value="shell" label="Shell">
+
+```shell
+curl -X PUT -H "Accept: application/vnd.gravitino.v1+json" \
+-H "Content-Type: application/json" -d '{
+    "privileges": [
+      {
+      "name": "SELECT_TABLE",
+      "condition": "ALLOW"
+      }]
+}' http://localhost:8090/api/metalakes/test/permissions/roles/role1/schema/catalog1.schema1/revoke
+
+curl -X PUT -H "Accept: application/vnd.gravitino.v1+json" \
+-H "Content-Type: application/json" -d '{
+    "privileges": [
+      {
+      "name": "SELECT_TABLE",
+      "condition": "ALLOW"
+      }]
+}' http://localhost:8090/api/metalakes/test/permissions/roles/role1/table/catalog1.schema1.table1/revoke
+```
+
+</TabItem>
+<TabItem value="java" label="Java">
+
+```java
+GravitinoClient client = ...
+        
+MetadataObject schema = ...
+Role role = client.grantPrivilegesToRole("role1", schema, Lists.newArrayList(Privileges.SelectTable.allow()));        
+        
+MetadataObject table = ...
+Role role = client.grantPrivilegesToRole("role1", table, Lists.newArrayList(Privileges.SelectTable.allow()));
```

Comment:
could you plz add comments to the demo code to make it more clearer and visual?

Reply:
Added.